libbeat/publisher/pipeline/module.go

```diff
@@ -157,6 +157,10 @@ func createQueueBuilder(config common.ConfigNamespace) (func(queue.Eventer) (que
 		queueConfig = common.NewConfig()
 	}
 
+	stateRegistry := monitoring.GetNamespace("state").GetRegistry()
+	queueRegistry := stateRegistry.NewRegistry("queue")
+	monitoring.NewString(queueRegistry, "name").Set(queueType)
```

Comment:
The pipeline should not use any globals at all, so one can safely instantiate multiple pipelines via Load. The Load function even gets the metrics registry passed as parameter (optional, can be nil).

Reply:
The registry passed is not the same into which this data has to go.

If there can be multiple queues in the future we would need a variable inside this package to keep track of it like we do for modules. Would that be ok for you?

Reply:
I understand it's a different regsitry. In this case we should pass a 'telemetry' registry or some kind of callback to the loader. But no globals. It's one of the few packages with no globals in mind from the very beginning.

Me might have other places where we need to pass a telemetry and a metrics registry to a feature/module/plugin. Maybe we want to pass a struct or interface of type:

```
type monitoring interface {
  Telemetry() *monitoring.Registry
  Metrics() *monitoring.Registry
  Logs() logp.Logger
}
```

Reply:
@ruflin I see some globals have been sneaked in... see #8091 for cleanups + introduction of queue type telemetry

Reply:
Yes.